taskiq/result_backends/redis/redis_backend.py

```diff
@@ -0,0 +1,93 @@
+import pickle
+from typing import Any, Dict, TypeVar
+
+from redis_client import RedisClient
+from taskiq.abc.result_backend import TaskiqResult
+
+from taskiq import AsyncResultBackend
+
+_ReturnType = TypeVar("_ReturnType")
+
+
+class RedisAsyncResultBackend(AsyncResultBackend[_ReturnType]):
+    """Async result based on redis."""
+
+    def __init__(self, redis_url: str):
+        self.redis_url = redis_url
+
+    async def startup(self) -> None:
+        """Makes redis connection on startup."""
+        self.redis_client = RedisClient(redis_url=self.redis_url)
+
+    async def shutdown(self) -> None:
+        """Closes redis connection."""
+        await self.redis_client.close()
+
+    async def set_result(
+        self,
+        task_id: str,
+        result: TaskiqResult[_ReturnType],
+    ) -> None:
+        """
+        Sets task result in redis.
+
+        Dumps TaskiqResult instance into the bytes and writes
+        it to redis.
+
+        :param task_id: ID of the task.
+        :param result: TaskiqResult instance.
+        """
+        to_insert_result_data = {}
+
+        for result_key, result_value in result.__dict__.items():
```

Comment:
Ok, maybe result must be a pydantic model, so you can easily cast it to dict. Got it.

Reply:
https://github.com/taskiq-python/taskiq/issues/12